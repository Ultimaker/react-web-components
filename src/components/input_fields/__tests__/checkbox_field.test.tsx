// Copyright (c) 2018 Ultimaker B.V.
import * as React from 'react';
import { shallow } from 'enzyme';

// component
import CheckboxField, {CheckboxFieldProps} from '../checkbox_field';
import Checkbox from '../../checkbox';
import InputFieldWrapper from '../input_field_wrapper';

describe('The checkbox field component', () => {
    let props: CheckboxFieldProps;
    let wrapper;

    beforeEach(() => {
        props = {
            id: 'testInputField',
            value: true,
            onChangeHandler: jest.fn(),
            labelLayoutWidth: '1/1',
            labelWidthBreakpoint: 'sm',
            staticField: false,
        };
        wrapper = shallow(<CheckboxField {...props} />);
    });

    it('should render a wrapper', () => {
        expect(wrapper).toMatchSnapshot();
        expect(props.onChangeHandler).not.toHaveBeenCalled();
    });

    it('should render a checkbox input', () => {
        expect(wrapper.find(Checkbox).props()).toEqual({
            disabled: false,
            id: "testInputField",
            onChangeHandler: wrapper.instance()._onChange,
            value: true,
        });
    })

    it('should call the callback', () => {
        expect(props.onChangeHandler).not.toHaveBeenCalled();
        wrapper.find(Checkbox).prop("onChangeHandler")(false)
        expect(props.onChangeHandler).toHaveBeenCalledWith(props.id, false);
        expect(wrapper.find(InputFieldWrapper).prop("touched")).toEqual(true);
    });
});
