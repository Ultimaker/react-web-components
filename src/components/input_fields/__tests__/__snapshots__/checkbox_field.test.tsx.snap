// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The checkbox field component should render a wrapper 1`] = `
<InputFieldWrapper
  className="input-field--checkbox"
  id="testInputField"
  labelLayoutWidth="1/1"
  labelWidthBreakpoint="sm"
  onChangeHandler={[MockFunction]}
  staticField={false}
  touched={false}
>
  <Checkbox
    disabled={false}
    id="testInputField"
    onChangeHandler={[Function]}
    value={true}
  />
</InputFieldWrapper>
`;
